lib/matplotlib/path.py

```diff
@@ -528,22 +587,39 @@ def contains_path(self, path, transform=None):
             transform = transform.frozen()
         return _path.path_in_path(self, None, path, transform)
 
-    def get_extents(self, transform=None):
+    def get_extents(self, transform=None, **kwargs):
         """
-        Return the extents (*xmin*, *ymin*, *xmax*, *ymax*) of the path.
+        Get Bbox of the path.
 
-        Unlike computing the extents on the *vertices* alone, this
-        algorithm will take into account the curves and deal with
-        control points appropriately.
+        Parameters
+        ----------
+        transform : matplotlib.transforms.Transform, optional
+            Transform to apply to path before computing extents, if any.
+        **kwargs
+            Forwarded to `.iter_bezier`.
+
+        Returns
+        -------
+        matplotlib.transforms.Bbox
+            The extents of the path Bbox([[xmin, ymin], [xmax, ymax]])
         """
         from .transforms import Bbox
-        path = self
         if transform is not None:
-            transform = transform.frozen()
-            if not transform.is_affine:
-                path = self.transformed(transform)
-                transform = None
-        return Bbox(_path.get_path_extents(path, transform))
+            self = transform.transform_path(self)
+        # return value for empty paths to match what used to be done in _path.h
+        extents = np.array([np.inf, np.inf, -np.inf, -np.inf])
+        for curve, code in self.iter_bezier(**kwargs):
+            # start and endpoints can be extrema of the curve
```

Comment:
perhaps the whole thing can be (approx)
```
for curve, _ in self.iter_bezier(...):
    _, dzeros = curve.axis_aligned_extrema
    potential_extrema = curve([0, *dzeros, 1])
```
?
(yes, you'd need to move `__call__` to this PR but I guess that's fine).

Reply:
On the other hand, this *greatly* improves readability, and I just didn't think to do it.

Changed a relevant docstring in `~.transforms.Bbox` to prevent someone from misunderstanding that spec in the future and breaking this code (since the spec was kinda ambiguously phrased before).

Bumped the addition of a vectorized `BezierSegment.__call__` to this PR so that the new code can work.

Reply:
As a side effect, doing this in a vectorized way makes using the standard Bbox methods faster than my custom _update_extents code again, which allows even further cleanup.